### PR TITLE
Use index.php for serving content

### PR DESCRIPTION
**Pull request by chillu**
See https://github.com/silverstripe/silverstripe-framework/issues/7419 and https://github.com/silverstripe/silverstripe-framework/issues/7430

### CODE REVIEW

**Comment by chillu** [comment]
Behat tests are failing until https://github.com/silverstripe/silverstripe-serve/pull/16 is merged. That change is backwards compatible with SS3 and the current SS4 (using `framework/main.php`), so can be merged prior to the other changes.

File: src/Control/HTTPRequestBuilder.php [hunk 1]
```diff
@@ -17,10 +17,13 @@ public static function createFromEnvironment()
     {
         // Clean and update live global variables
         $variables = static::cleanEnvironment(Environment::getVariables());
-        Environment::setVariables($variables); // Currently necessary for SSViewer, etc to work
 
         // Health-check prior to creating environment
-        return static::createFromVariables($variables, @file_get_contents('php://input'));
+        $req = static::createFromVariables($variables, @file_get_contents('php://input'));
+
+        Environment::setVariables($variables); // Currently necessary for SSViewer, etc to work
```
Comment (tractorcow) [inline src/Control/HTTPRequestBuilder.php]: Not sure why this order had to be switched. createFromVariables() should not be modifying $variables...

**Comment by tractorcow** [comment]
This bit is incorrect;

```

**Comment by tractorcow** [comment]
// Remove query parameters (they're retained separately through $server['_GET']
$url = parse_url($variables['_SERVER']['REQUEST_URI'], PHP_URL_PATH);
 // Remove base folders from the URL if webroot is hosted in a subfolder
if (substr(strtolower($url), 0, strlen(BASE_URL)) === strtolower(BASE_URL)) {
    $url = substr($url, strlen(BASE_URL));
}
 // Normalise URI
$variables['_SERVER']['REQUEST_URI'] = $url;
```

**Comment by chillu** [comment]
> $_SERVER['REQUEST_URI'] needs to be root-relative; E.g. /basedir/page not /page; Otherwise it breaks sites running in sub-folders.

Good catch, looks like you've fixed that already. I've checked operation in a subfolder, works fine (as long as you don't define `SS_BASE_URL` - which the installer happens to do, and that skewed my testing and sent me on a wild goose chase around core grrrr!)

**Comment by tractorcow** [comment]
Yes I've done that before. :)

**Pull request merged**